code/modules/station_goals/bsa.dm

```diff
@@ -363,7 +363,10 @@ GLOBAL_VAR_INIT(bsa_unlock, FALSE)
 		notice = "Cannon unpowered!"
 		return
 	notice = null
-	cannon.fire(user, get_impact_turf())
+	var/turf/target_turf = get_impact_turf()
+	if(obj_flags & EMAGGED)
+		target_turf = get_turf(src)
```

Comment:
shouldnt this behavior be in get_impact_turf ??

Reply:
fixed